testing/upgradetest/upgrader.go

```diff
@@ -250,6 +259,13 @@ func PerformUpgrade(
 	}
 	logger.Logf("upgrade watcher started")
 
+	// Check that, while the Upgrade Watcher is running, the upgrade details in Agent status
+	// show the state as UPG_WATCHING.
+	if err := checkUpgradeDetailsState(ctx, startFixture, details.StateWatching); err != nil {
+		// error context added by checkUpgradeDetailsState
+		return err
+	}
+
```

Comment:
Maybe we want to check for a final state of the upgrade like COMPLETED or FAILED ?